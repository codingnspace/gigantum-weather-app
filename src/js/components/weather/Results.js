//Vendor
import React, {Component} from 'react';
import PropTypes from 'prop-types';
//Locals
import bike from 'images/bicycle.svg';
import metro from 'images/metro.svg';

class Results extends Component {
  static propTypes ={
    results: PropTypes.array
  }

  static defaultProps = {
    results: []
  }

  render () {
    const resultList = this.props.results && this.props.results.map((result, idx) => {
      return (
        <div key={result.idx} className="result">
          <div className='result-day-text'>{idx === 0 ? 'Today' : result.day}</div>
          <div className='result-img'>
            <img src={result.decision === 'bike' ? bike : metro} alt={result.decision === 'bike' ? 'bike_img' : 'metro_img'} />
          </div>
        </div>
      );
    });
    return (
      <div className="result-list">
        <h2>Your Results Are In...</h2>
        {resultList}
        <div className="btn-wrapper">
          <button onClick={this.props.resetResults}>Start Over</button>
        </div>
      </div>
    );
  }
}

export default Results;
