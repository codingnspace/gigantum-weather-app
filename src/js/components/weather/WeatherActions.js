import {getApiData} from 'js/utils/api';
import {testUrl} from 'js/config';

export function getWeather() {
  return (dispatch, getState) => {
    // const something = api();
    fetch(testUrl)
    .then((response) => {
      return response.json();
    })
    .then((response) => {
      console.log(response);
    })
    .catch((error) => {
      console.log(error);
    });
  };
}
