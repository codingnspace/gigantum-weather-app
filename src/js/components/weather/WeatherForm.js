//Vendor
import React, {Component} from 'react';
import PropTypes from 'prop-types';
import { Field, reduxForm } from 'redux-form';
//Locals
import temperature from 'images/temperature.svg';
import percent from 'images/percent.svg';

class WeatherForm extends Component {
  static propTypes = {
    handleSubmit: PropTypes.func,
    getWheather: PropTypes.func
  }

  handleSubmit = (e) => {
    e.preventDefault();
    this.props.handleSubmit((formValues) => {
      this.props.getWeather(formValues);
    })();
  }

  render () {
    return (
      <div className="weather-form-container">
        <h2>Put In Your Conditions</h2>
        <form onSubmit={this.handleSubmit} className="weather-form">
            <div className="input-with-icon">
              <Field component="input" placeholder="Min Temp" name="minTemp" />
              <img className="app-logzzo" src={temperature} alt="app-logo" />
            </div>
            <div className="input-with-icon">
              <Field component="input" placeholder="Max Temp" name="maxTemp" />
              <img className="app-logzzo" src={temperature} alt="app-logo" />
            </div>
            <div className="input-with-icon">
              <Field component="input" placeholder="Max Rain Percentage" name="maxRain" />
              <img className="app-logzzo" src={percent} alt="app-logo" />
            </div>
            <div className="btn-wrapper">
              <button type="submit">Get Info</button>
            </div>
        </form>
      </div>
    );
  }
}

export default reduxForm({form: 'weather'})(WeatherForm);
