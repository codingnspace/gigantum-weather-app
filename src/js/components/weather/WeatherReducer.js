// import Immutable from 'immutable';
export const weatherReducer = (state = {}, action) => {
  switch (action.type) {
    case 'SET_VISIBILITY_FILTER':
      return action.filter;
    default:
      return state;
  }
};
