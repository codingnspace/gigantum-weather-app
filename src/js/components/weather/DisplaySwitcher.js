//Vendor
import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {sortBy} from 'lodash';
//Locals
import WeatherForm from './WeatherForm';
import Results from './Results';
import {getWeather, resetDecision} from 'js/components/weather/WeatherActions';

class DisplaySwitcher extends Component {
  static propTypes ={
    getWheather: PropTypes.func,
    results: PropTypes.array,
    resetDecision: PropTypes.func
  }

  render () {
    const {results} = this.props;
    let display;
    if (results && results.length > 0) {
      display = <Results results={results} resetResults={this.props.resetDecision} />;
    } else {
      display = <WeatherForm getWeather={this.props.getWeather} />;
    }
    return (
      <div className="display-switcher">
        {display}
      </div>
    );
  }
}

function mapStateToProps(state) {
  return {
    results: sortBy(state.weatherReducer, 'idx')
  };
}
export const DisconnectedDisplaySwiter = DisplaySwitcher;
export default connect(mapStateToProps, {
  getWeather,
  resetDecision
})(DisplaySwitcher);
