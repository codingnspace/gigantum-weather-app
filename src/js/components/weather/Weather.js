//Vendor
import React, {Component} from 'react';
import {connect} from 'react-redux';
import {compose} from 'recompose';
import { Field, reduxForm } from 'redux-form';
//Locals
import {getWeather} from 'js/components/weather/WeatherActions';

class Weather extends Component {
  handleSubmit = (e) => {
    e.preventDefault();
    this.props.getWeather();
    this.props.handleSubmit((formValues) => {
      console.log(formValues);
    })();
  }

  render () {

    return (
      <form onSubmit={this.handleSubmit}>
        <Field component="input" placeholder="Min Temp" name="minTemp" /><br />
        <Field component="input" placeholder="Max Temp" name="maxTemp" /><br />
        <Field component="input" placeholder="Max Rain Percentage" name="maxRainChance" />
        <button type="submit">submit</button>
      </form>
    );
  }
}

export default compose(
  reduxForm({form: 'weather'}),
  connect(null, getWeather)
)(Weather);
