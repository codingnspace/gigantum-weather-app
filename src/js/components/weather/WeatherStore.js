import {weatherReducer} from 'js/components/weather/WeatherReducer';
import {combineReducers, createStore, applyMiddleware} from 'redux';
import thunk from 'redux-thunk';
import {reducer as formReducer} from 'redux-form';
import {compose} from 'recompose';

const rootReducer = combineReducers({
  form: formReducer,
  weatherReducer
});

const enhancers = compose(
  applyMiddleware(thunk),
  window.devToolsExtension ? window.devToolsExtension() : f => f
);

export const weatherStore = createStore(rootReducer, {}, enhancers);
