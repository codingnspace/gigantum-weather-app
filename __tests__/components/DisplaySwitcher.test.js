import { shallowToJson } from 'enzyme-to-json';
import { configure, shallow } from 'enzyme';
import React from 'react';

import {DisconnectedDisplaySwiter} from 'js/components/weather/DisplaySwitcher';
import Adapter from 'enzyme-adapter-react-15';

configure({adapter: new Adapter()});

describe('DisplaySwitcher', () => {

  it('should display the Results component when results prop array has more than 0 items', () => {
    const results = [{
      idx: 0,
      day: 'Monday',
      decision: 'metro'
    }];
    const output = shallow(
      <DisconnectedDisplaySwiter results={results} getWeather={()=>null} />
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });

  it('should display the WeatherForm component when results prop array has 0 items', () => {
    const results = [];
    const output = shallow(
      <DisconnectedDisplaySwiter results={results} getWeather={()=>null} />
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });

});
