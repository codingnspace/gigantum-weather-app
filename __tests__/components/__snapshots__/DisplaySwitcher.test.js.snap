// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisplaySwitcher should display the Results component when results prop array has more than 0 items 1`] = `
<div
  className="display-switcher"
>
  <Results
    results={
      Array [
        Object {
          "day": "Monday",
          "decision": "metro",
          "idx": 0,
        },
      ]
    }
  />
</div>
`;

exports[`DisplaySwitcher should display the WeatherForm component when results prop array has 0 items 1`] = `
<div
  className="display-switcher"
>
  <ReduxForm
    getWeather={[Function]}
  />
</div>
`;

exports[`DisplaySwitcher should render correctly 1`] = `
<div
  className="display-switcher"
>
  <Results
    results={
      Array [
        Object {
          "day": "Monday",
          "decision": "metro",
          "idx": 0,
        },
      ]
    }
  />
</div>
`;

exports[`DisplaySwitcher should render correctly 2`] = `
<div
  className="display-switcher"
>
  <ReduxForm
    getWeather={[Function]}
  />
</div>
`;
